### ISSUE TITLE
tf_text_graph_ssd.py produce invalid text-graph representation

### ISSUE THREAD

**Issue by roccopietrini** [opencv/opencv#17484]
##### System information (version)
- OpenCV => 4.3.0

##### Detailed description

Hi, I'm trying to generate the text graph representation for MobileNetV3-SSD from tensorflow to be used in DNN module. I fine-tuned the ssd_mobilenet_v3_small_coco_2020_01_14 on a custom class and everything was fine in the text graph generation, everything is also working with the DNN module. 
Then I resumed training modifying the pipeline.config again with some data augmentation and dropout. When I genereate the text-graph representation I have no erros.. but then when I import with

`net = cv.dnn_DetectionModel('frozen_inference_graph.pb', 'cvgraph.pbtxt')`

this error appear:

"Input layer not found: FeatureExtractor/MobilenetV3/expanded_conv_8/expansion_output in function 'connect"

The same network, with different pipeline.config generate two different text graph representations, but the second one doesn't even work with DNN module. Strange thing is that if I use the second frozen_inference_graph with the first cvgraph.pbtxt everything works! 

 
[pipeline.zip](https://github.com/opencv/opencv/files/4737303/pipeline.zip)

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       answers.opencv.org, Stack Overflow, etc and have not found solution
   
 - [x] I updated to latest OpenCV version and the issue is still there
   
 - [x] There is reproducer code and related data files: videos, images, onnx, etc

**Comment by sergregory** [comment]
may be similar to https://github.com/opencv/opencv/issues/17364

**Comment by LupusSanctus** [comment]
@roccopietrini, if you have `frozen_inference_graph.pb` file for fine-tuned  `ssd_mobilenet_v3_small_coco_2020_01_14`  and/or obtained text graphs, please, attach them to the issue.
I there is no possibility to attach these files, could you, please, provide some more details about `ssd_mobilenet_v3_small_coco_2020_01_14` fine-tuning? Which steps exactly were passed to obtain 'frozen_inference_graph.pb'? Did you use [TF Object Detection](https://github.com/tensorflow/models/tree/master/research/object_detection) for this purpose, for example:
1. training:
```
python model_main_tf2.py --model_dir="training_dir" --pipeline_config_path="pipeline.config"
```
or TF 1.5

```
python object_detection/model_main.py --model_dir="training_dir" --pipeline_config_path="pipeline.config"
```

2. extracting the model:
```
python exporter_main_v2.py --input_type=image_tensor --pipeline_config_path="pipeline.config" --trained_checkpoint_dir="training_dir" --output_directory="training_dir/model"
```

or TF 1.5

```
python export_inference_graph.py --input_type=image_tensor --pipeline_config_path="pipeline.config --output_directory="training_dir/model" --trained_checkpoint_prefix=<...>
```
?

**Comment by roccopietrini** [comment]
Hi, it's been long time and I don't have that frozen_inference_graph.pb anymore, but the pipeline.config used for that training is the one attached above. The problem was that with the same dataset, only changing some minor training parameters (like adding data augmentations) the script produced an invalid text-graph representation. I used TF 1.5 like you said with TFOD API. I'll try next days again to find this stuff and report back if anything relevant appears.

**Comment by LupusSanctus** [comment]
@roccopietrini, thank you a lot!
